docs: add docs for Python runtime (#1549)
keptn/lifecycle-toolkit: docs/content/en/docs/implementing/tasks/python.md

---
title: Working with python runtime for Keptn tasks
description: Learn how to work with Keptn tasks using python
weight: 90
hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html

---

[KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md/)

> resource. A task definition can use a python runtime to specify a task that runs python3 code.

Reviewer: ```suggestion
resource.
A task definition can use a python runtime to specify a task that runs python3 code.
```

---
title: Working with python runtime for Keptn tasks
description: Learn how to work with Keptn tasks using python
weight: 90
hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html

---

[KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md/)
resource.
A task definition can use a python runtime to specify a task that runs python3 code.

> The python code can be specified as: 
> - inline

Reviewer: ```suggestion
The python code can be specified as: 

- inline
```

---
title: Working with python runtime for Keptn tasks
description: Learn how to work with Keptn tasks using python
weight: 90
hidechildren: false # this flag hides all sub-pages in the sidebar-multicard.html

---
Keptn tasks are defined in a
[KeptnTaskDefinition](../../yaml-crd-ref/taskdefinition.md/)
resource.

> A task definition can use a python runtime to specify a task that runs python3 code.
> The python code can be specified as:

Reviewer 2: ```suggestion
A task definition can use the Python runtime to specify a task that runs Python 3 code.
The Python code can be specified as:
```

- inline
- an HTTP reference to a script
- another `KeptnTaskDefinition`
- a [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/)
  resource that is populated with the code to execute

## Inline

> You can embed python code directly in the taskDefinition.

Reviewer 2: either use `KeptnTaskDefinition` or write it in separate words
```suggestion
You can embed python code directly in the task definition.
```

Consider the following example, where we print data stored in the parameters map:
{{< readfile file="/yaml/python_execution/taskdefinition_pyfunction_inline.yaml" code="true" lang="yaml" >}}

> ## Http reference

Reviewer 2: ```suggestion
## HTTP reference
```

You can refer to a code stored online, for instance we have a few examples available [here](https://github.com/keptn/lifecycle-toolkit/tree/main/python-runtime/samples).
Consider the following:
{{< readfile file="/yaml/python_ execution/taskdefinition_pyfunction_configmap.yaml" code="true" lang="yaml" >}}

## Refer to another TaskDefinition

> You can refer to an existing taskDefinition. 

Reviewer 2: ```suggestion
You can refer to an existing `KeptnTaskDefinition`. 
```

Consider the following example, this will call the inline example but will override the data printed with what is specified in the task:
{{< readfile file="/yaml/python_ execution/taskdefinition_pyfunction_recursive.yaml" code="true" lang="yaml" >}}

## Config Map

Consider the following example:
{{< readfile file="/yaml/python_ execution/taskdefinition_pyfunction_configmap.yaml" code="true" lang="yaml" >}}

## Allowed libraries for the Python runtime

The following example shows how to use few of the allowed packages, namely: requests, json, git, yaml
{{< readfile file="/yaml/python_ execution/taskdefinition_pyfunction_inline_printargs_py.yaml" code="true" lang="yaml" >}}

> ## Passing secrets, env vars and modifying the python command

Reviewer 2: ```suggestion
## Passing secrets, environment variables and modifying the python command
```

---

> title: Working with Python runtime for Keptn tasks

Reviewer: This file should be removed from this PR.  The text was stolen and incorporated into the implementing/tasks file in https://github.com/keptn/lifecycle-toolkit/pull/1521
